### ISSUE TITLE
Question regarding the filtering step with preFilter()

### ISSUE THREAD

**Issue by rauldiul** [kvittingseerup/IsoformSwitchAnalyzeR#115]
Hi,

thank you for your great package. I have a simple question regarding the filtering done with the **preFilter** function, which I would like to understand better.

I have a SwitchList with 97214 isoforms, which I filter like this:
```

SwitchList.f <- preFilter(
  switchAnalyzeRlist = SwitchList,
  geneExpressionCutoff = 1,
  isoformExpressionCutoff = 0,
  IFcutoff = 0.01,
  removeSingleIsoformGenes = TRUE,
  quiet = FALSE
)

The filtering removed 49816 ( 51.24% of ) transcripts. There is now 47398 isoforms left
```

However, when I compare the original and the filtered objects, I see different sizes for some of the matrices within the filtered object: 47398 versus 68360

```
unlist(lapply(SwitchList,nrow)); unlist(lapply(SwitchList.f,nrow))
     isoformFeatures           conditions         designMatrix   isoformCountMatrix isoformRepExpression          orfAnalysis         isoformRepIF 
              583284                    4                   12                97214                97214                97214                97214 
     isoformFeatures           conditions         designMatrix   isoformCountMatrix isoformRepExpression          orfAnalysis         isoformRepIF 
              284388                    4                   12                68360                68360                47398                68360 
```

I understand that the "definitive" set of analyzed isoforms are those in the **orfAnalysis** matrix, right? Because those are 47398, and that number was printed by the verbose in the **preFilter** function. Also, in my case I have 6 comparisons and comparing the rows with that matrix I get 6:

```
nrow(SwitchList.f$isoformFeatures) / nrow(SwitchList.f$orfAnalysis)
[1] 6
```

Why do the count and abundance matrices have more isoforms in the filtered object? Is it to be able to compute the IF for the isoforms which are finally used? I suspect this because if I disable the filtering based on IFcutoff, I am able to get the same numbers of isoforms across the matrices:

```
SwitchList.f <- preFilter(
  switchAnalyzeRlist = SwitchList,
  geneExpressionCutoff = 1,
  isoformExpressionCutoff = 0,
  IFcutoff = -1,
  removeSingleIsoformGenes = TRUE,
  quiet = FALSE
)
unlist(lapply(SwitchList,nrow)); unlist(lapply(SwitchList.f,nrow))
     isoformFeatures           conditions         designMatrix   isoformCountMatrix isoformRepExpression          orfAnalysis         isoformRepIF 
              583284                    4                   12                97214                97214                97214                97214 
     isoformFeatures           conditions         designMatrix   isoformCountMatrix isoformRepExpression          orfAnalysis         isoformRepIF 
              431202                    4                   12                71867                71867                71867                71867 
```

So, to sum up, **are the 47398 isoforms in orfAnalysis the actual isoforms which will be tested subsequently by DEXseq, and why are there more isoforms in the count matrices?**

thanks for your help and your time!

**Comment by kvittingseerup** [comment]
You are right. To ensure gene abundance (and thereby also IF calculations) are correct all isoforms from genes analyzed are kept.

Only isoforms in `<your_switchAnalyzeRlist>$isoformFeatures` are analyzed.
